src/styles/_catalog.scss

```diff
@@ -1,22 +1,35 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
-
-  max-width: 944px;
-  margin: 0 auto;
+  display: grid;
+  justify-content: center;
+  gap: 48px;
+  grid-template-columns: repeat(1,$card-width);
 
   &__card {
     display: flex;
     justify-content: center;
     align-items: center;
-    width: 200px;
+    width: $card-width;
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+}
+
+@media  (min-width: 488px) {
+  .catalog {
+    grid-template-columns: repeat(2,$card-width);
+  }
+}
+
+@media  (min-width: 768px) {
+  .catalog {
+    grid-template-columns: repeat(3,$card-width);
+  }
+}
+
+@media  (min-width: 1024px) {
+  .catalog {
+    grid-template-columns: repeat(4,$card-width);
```

Comment:
I recomended  you move your @media blocks to class .catalog (before class .card) and delete ".catalog {" intro @media